pkg/services/dashboards/accesscontrol.go

```diff
@@ -173,15 +173,14 @@ func resolveDashboardScope(ctx context.Context, db Store, folderDB FolderStore,
 	return result, nil
 }
 
-func getInheritedScopes(ctx context.Context, orgID int64, folderUID string, folderSvc folder.Service) ([]string, error) {
+func GetInheritedScopes(ctx context.Context, orgID int64, folderUID string, folderSvc folder.Service) ([]string, error) {
 	ancestors, err := folderSvc.GetParents(ctx, folder.GetParentsQuery{
 		UID:   folderUID,
 		OrgID: orgID,
 	})
 
 	if err != nil {
-		// TODO return a specific error
-		return nil, err
+		return nil, folder.ErrInternal.Errorf("could not retrieve folder parents: %w", err)
```

Comment:
I would recommend defining a new error inside the accesscontrol service instead so that from the log entry we can easier spot the source of the problem.

Reply:
Makes sense. I was thinking about it as a folder error, but you're right that this function is only used by access control, so it makes more sense for it to return an access control error.